creator/data_templates/mutations/template_version.py

```diff
@@ -0,0 +1,285 @@
+        "organization's studies. This will override anything in the studies "
+        "input parameter.",
+    )
+    studies = graphene.List(
+        graphene.ID,
+        description="The studies this template version should be assigned to",
+    )
+
+
+class UpdateTemplateVersionInput(graphene.InputObjectType):
+    """Parameters used when updating an existing template_version"""
+
+    description = graphene.String(
+        description="The description of the template_version"
+    )
+    field_definitions = graphene.JSONString(
+        description="The field definitions for this template"
+    )
+    apply_to_all = graphene.Boolean(
+        description="Whether to add this template version to all of the "
+        "organization's studies. This will override anything in the studies "
+        "input parameter.",
+    )
+    studies = graphene.List(
+        graphene.ID,
+        description="The studies this template version should be assigned to",
+    )
+
+
+class CreateTemplateVersionMutation(graphene.Mutation):
+    """Creates a new template_version"""
+
+    class Arguments:
+        input = CreateTemplateVersionInput(
+            required=True,
+            description="Attributes for the new template_version",
+        )
+
+    template_version = graphene.Field(TemplateVersionNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new template_version.
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.add_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        # Check if data template exists
+        dt_id = input.pop("data_template")
+        model, dt_id = from_global_id(dt_id)
+
+        try:
+            dt = DataTemplate.objects.get(pk=dt_id)
+        except DataTemplate.DoesNotExist:
+            raise GraphQLError(f"DataTemplate {dt_id} does not exist")
+
+        # User may only add template versions to a template owned by their org
+        if not user.organizations.filter(pk=dt.organization.pk).exists():
+            raise GraphQLError(
+                "Not allowed - may only add template versions to templates "
+                "that are owned by the user's organization."
+            )
+
+        # Get apply flag for later
+        apply_to_all = input.pop("apply_to_all", None)
+
+        # Get study ids for later
+        study_ids = input.pop("studies", None)
+
+        # Create template version
+        with transaction.atomic():
+            template_version = TemplateVersion(**{k: input[k] for k in input})
+            template_version.creator = user
+            template_version.data_template = dt
+
+            # Validate and save template
+            template_version.clean()
+            template_version.save()
+
+            # Add studies to the template version if they exist
+            if apply_to_all:
+                template_version.studies.set(dt.organization.studies.all())
```

Comment:
Do you need to `check_studies` first?

Reply:
Ah yeaaa, and I think may even need to modify `check_studies`. I just realized that it currently only checks if the user is a member of the studies that are being mutated. But maybe it should check whether the user has an admin permission (like change_study? I forget what its called) or if they are a member of the study?